test-app/tests/integration/components/input-field-test.gts

```diff
@@ -0,0 +1,166 @@
+    assert.dom(input).hasAttribute('aria-invalid', 'true');
+  });
+
+  test('it renders hint text', async function (assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        @hint="Hint text visible here"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+    const hint = '[data-hint]';
+
+    assert.dom(label).exists('Expected to have label component rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "label"');
+
+    assert.dom(input).exists('Expected to have input tag rendered');
+    assert.dom(input).hasAttribute('type', 'text');
+   
+    assert.dom(hint).exists('Expected to have hint component rendered');
+    assert.dom(hint).hasText('Hint text visible here', 'Expected to have hint text "error"');
+    assert.dom(hint).hasAttribute('id');
+    const hintId = find(hint)?.getAttribute('id') || '';
+    const describedby = find(input)?.getAttribute('aria-describedby') || '';
+
+    assert.ok(describedby.includes(hintId), 'Expected hintId to be included in the aria-describedby');
+  });
+
+  test('it sets aria-describedby when both a hint and error are provided using the hint and error ids', async function(assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        @hint="Hint text visible here"
+        @error="Error text"
+        data-input
+        />
+    </template>);
+
+    const errorId = find('[data-error]')?.getAttribute('id') || '';
+    assert.ok(errorId, 'Expected errorId to be truthy');
+
+    const hintId = find('[data-hint]')?.getAttribute('id') || '';
+    assert.ok(hintId, 'Expected hintId to be truthy');
+
+    assert.dom('[data-input]').hasAttribute('aria-describedby', `${hintId} ${errorId}`);
+  });
+
+  test('it accepts @value and @onChange', async function (assert) {
+    assert.expect(8);
+    
+    const onChangeCallback = (value: string, e: Event | InputEvent) => {
+      assert.strictEqual(value, 'Banana', 'Expected input to match');
+      assert.ok(e, 'Expected `e` to be available as the second argument');
+      assert.ok(e.target, 'Expected direct access to target from `e`');
+      assert.step('handleChange');
+    };
+  
+    await render(<template>
+        <InputField
+          @label="Label"
+          type="text"
+          @value="Avocado"
+          @onChange={{onChangeCallback}}
+          data-input
+          />
+      </template>);
+
+    assert.verifySteps([]);
+
+    const input: HTMLInputElement | null  = find('[data-input]') as HTMLInputElement;
+
+    if (!input) {
+      throw new Error('an input was not detected');
+    }
+   
+    assert.dom(input).hasValue('Avocado', 'input has the original value');
+
+    await fillIn('[data-input]', 'Banana');
+
+    assert.dom(input).hasValue('Banana', 'input has the set @value');
+
+    assert.verifySteps(['handleChange']);
```

Comment:
Nit / no need to change: this could maybe be simplified with:

```suggestion
    assert.dom('[data-input]').hasValue('Avocado', 'input has the original value');

    await fillIn('[data-input]', 'Banana');

    assert.dom('[data-input]').hasValue('Banana', 'input has the set @value');

    assert.verifySteps(['handleChange']);
```

If `fillIn` doesn't find an element or if the value isn't set as expected you'd get an assertion error